tarantool-lsp/websocket-lib.lua

```diff
@@ -0,0 +1,97 @@
+local rpc             = require 'tarantool-lsp.rpc'
+local method_handlers = require 'tarantool-lsp.methods'
+local ws              = require 'websocket'
+local json              = require 'json'
+local docs 	          = require 'tarantool-lsp.tnt-doc.doc-manager'
+
+local handshake = require('websocket.handshake')
+local fio = require('fio')
+local log = require('log')
+
+--_G.Globals = _G.Globals or nil -- defined in analyze.lua
+--_G.Types = _G.Types or {}
+
+local DETACHED = 101
+
+local initalized = false
+
+local create_handler = function(options)
+    local path = debug.getinfo(1).source:match("@?(.*/)")
+
+    log.info('Path: %s', path)
+
+    if not initalized then
+        local ok, err = docs:init({ completions_dir = fio.pathjoin(path, 'completions') })
+        if err ~= nil then
+            log.info("Docs subsystem error: %s", err)
+        end
+        initalized = true
+    end
+    return function(req)
+        local validated, errresp = handshake.validate_request(req)
+
+        if validated then
+            --log.info(req)
+            req.s:write(
+                handshake.reduce_response(
+                    handshake.accept_upgrade(req)))
+            local ws_peer = ws.new(req.s, 10, false, true, req)
+            local config = {
+                language = "5.1",
+                builtins = {"5_1"},
+                packagePath = {"./?.lua"},
+                debugMode = false,
+                documents = {},
+                types = {},
+                web_server = true,
+                _useNativeLuacheck = false -- underscore means "experimental" here
+            }
+            while true do
```

Comment:
Do you have a way to generalize this loop?

Reply:
Yes  but think it's more for websocket library than for this library